x-pack/plugins/apm/server/lib/helpers/get_apm_alerts_client.ts

```diff
@@ -26,17 +28,19 @@ export async function getApmAlertsClient({
     throw Error('No alert indices exist for "apm"');
   }
 
-  type ApmAlertsClientSearchParams = Omit<
-    Parameters<typeof alertsClient.find>[0],
-    'index'
-  >;
+  type RequiredParams = ESSearchRequest & {
+    size: number;
+    track_total_hits: boolean | number;
+  };
 
   return {
-    search(searchParams: ApmAlertsClientSearchParams) {
+    search<TParams extends RequiredParams>(
+      searchParams: TParams
+    ): Promise<InferSearchResponseOf<ParsedTechnicalFields, TParams>> {
       return alertsClient.find({
         ...searchParams,
         index: apmAlertsIndices.join(','),
```

Comment:
It hit me: why are we even using the alertsClient and not just the vanilla esClient? Afaict all we need is the alert indices (`apmAlertsIndices`) from the alerting plugin.

It seems like ` alertsClient.find` is an abstraction on top of `esClient.search`. And not a particular nice one. Can you ask the response ops team if there are any downsides to using the vanilla es client?

```suggestion
      return esClient.search({
        ...searchParams,
        index: apmAlertsIndices.join(','),
```

Reply:
I will proceed with the merge in favour of FF but will start a slack thread separately.